cmake/Modules/FindTPLCUDA.cmake

```diff
@@ -7,37 +7,39 @@ IF (NOT CUDAToolkit_ROOT)
   ENDIF()
 ENDIF()
 
-# FIXME CMake 3.28.4 creates more targets than we export
-IF(CMAKE_VERSION VERSION_GREATER_EQUAL "3.17.0" AND CMAKE_VERSION VERSION_LESS "3.28.4")
-  find_package(CUDAToolkit)
+IF(CMAKE_VERSION VERSION_GREATER_EQUAL "3.17.0")
+  find_package(CUDAToolkit REQUIRED)
+  KOKKOS_CREATE_IMPORTED_TPL(CUDA INTERFACE
+    LINK_LIBRARIES CUDA::cuda_driver CUDA::cudart
+  )
+  KOKKOS_EXPORT_CMAKE_TPL(CUDAToolkit REQUIRED)
 ELSE()
   include(${CMAKE_CURRENT_LIST_DIR}/CudaToolkit.cmake)
-ENDIF()
-
 
-IF (TARGET CUDA::cudart)
-  SET(FOUND_CUDART TRUE)
-  KOKKOS_EXPORT_IMPORTED_TPL(CUDA::cudart)
-ELSE()
-  SET(FOUND_CUDART FALSE)
-ENDIF()
+  IF (TARGET CUDA::cudart)
+    SET(FOUND_CUDART TRUE)
+    KOKKOS_EXPORT_IMPORTED_TPL(CUDA::cudart)
+  ELSE()
+    SET(FOUND_CUDART FALSE)
+  ENDIF()
 
-IF (TARGET CUDA::cuda_driver)
-  SET(FOUND_CUDA_DRIVER TRUE)
-  KOKKOS_EXPORT_IMPORTED_TPL(CUDA::cuda_driver)
-ELSE()
-  SET(FOUND_CUDA_DRIVER FALSE)
-ENDIF()
+  IF (TARGET CUDA::cuda_driver)
+    SET(FOUND_CUDA_DRIVER TRUE)
+    KOKKOS_EXPORT_IMPORTED_TPL(CUDA::cuda_driver)
+  ELSE()
+    SET(FOUND_CUDA_DRIVER FALSE)
+  ENDIF()
 
-include(FindPackageHandleStandardArgs)
-IF(KOKKOS_CXX_HOST_COMPILER_ID STREQUAL NVHPC)
-  SET(KOKKOS_CUDA_ERROR "Using NVHPC as host compiler requires at least CMake 3.20.1")
-ELSE()
-  SET(KOKKOS_CUDA_ERROR DEFAULT_MSG)
-ENDIF()
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(TPLCUDA ${KOKKOS_CUDA_ERROR} FOUND_CUDART FOUND_CUDA_DRIVER)
-IF (FOUND_CUDA_DRIVER AND FOUND_CUDART)
-  KOKKOS_CREATE_IMPORTED_TPL(CUDA INTERFACE
-    LINK_LIBRARIES CUDA::cuda_driver CUDA::cudart
-  )
+  include(FindPackageHandleStandardArgs)
+  IF(KOKKOS_CXX_HOST_COMPILER_ID STREQUAL NVHPC)
+    SET(KOKKOS_CUDA_ERROR "Using NVHPC as host compiler requires at least CMake 3.20.1")
```

Comment:
This seems a bit weird. So, if you are using NVHPC and CMake version between 3.17 and 3.20.1, what happens? Would the stuff inside the `if()` clause work correctly in that case?

Reply:
Fixed in 9bf763d32.

Reply:
> Fixed in [9bf763d](https://github.com/kokkos/kokkos/commit/9bf763d32dd1e6e9b2ff459a5136e6afea7683d9).

I'm confused. Isn't `KOKKOS_CUDA_ERROR` only used in the `else()` cause? I was just wondering about what happens in that case in the `if()` branch, not the placement of that code.